java/src/main/native/src/ColumnViewJni.cpp

```diff
@@ -0,0 +1,1655 @@
+      rmm::device_buffer null_mask{0};
+      copy->set_null_mask(null_mask);
+      return reinterpret_cast<jlong>(copy.release());
+    }
+
+    std::vector<cudf::column_view> column_views;
+    std::transform(n_cudf_columns.data(), n_cudf_columns.data() + n_cudf_columns.size(),
+                   std::back_inserter(column_views),
+                   [](auto const &p_column) { return *p_column; });
+    cudf::table_view *input_table = new cudf::table_view(column_views);
+
+    cudf::binary_operator op = static_cast<cudf::binary_operator>(bin_op);
+    if(op == cudf::binary_operator::BITWISE_AND) {
+      copy->set_null_mask(cudf::bitmask_and(*input_table));
+    }
+
+    return reinterpret_cast<jlong>(copy.release());
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_hash(JNIEnv *env,
+                                                            jobject j_object,
+                                                            jlongArray column_handles,
+                                                            jint hash_function_id) {
+  JNI_NULL_CHECK(env, column_handles, "array of column handles is null", 0);
+
+  try {
+    cudf::jni::native_jpointerArray<cudf::column_view> n_cudf_columns(env, column_handles);
+    std::vector<cudf::column_view> column_views;
+    std::transform(n_cudf_columns.data(), n_cudf_columns.data() + n_cudf_columns.size(),
+                   std::back_inserter(column_views),
+                   [](auto const &p_column) { return *p_column; });
+    cudf::table_view *input_table = new cudf::table_view(column_views);
+
+    std::unique_ptr<cudf::column> result = cudf::hash(*input_table, static_cast<cudf::hash_id>(hash_function_id));
+    return reinterpret_cast<jlong>(result.release());
+  }
+  CATCH_STD(env, 0);
+}
+
+////////
+// Native cudf::column_view life cycle and metadata access methods. Life cycle methods
+// should typically only be called from the CudfColumn inner class.
+////////
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_makeCudfColumnView(JNIEnv *env,
+                                                                          jclass, jint j_type,
+                                                                          jint scale, jlong j_data,
+                                                                          jlong j_data_size,
+                                                                          jlong j_offset,
+    jlong j_valid, jint j_null_count, jint size, jlongArray j_children) {
+
+  try {
+    using cudf::column_view;
+    cudf::jni::auto_set_device(env);
+    cudf::type_id n_type = static_cast<cudf::type_id>(j_type);
+    cudf::data_type n_data_type = cudf::jni::make_data_type(j_type, scale);
+
+    std::unique_ptr<cudf::column_view> ret;
+    void *data = reinterpret_cast<void *>(j_data);
+    cudf::bitmask_type *valid = reinterpret_cast<cudf::bitmask_type *>(j_valid);
+    cudf::jni::native_jlongArray retTmp(env, 2);
```

Comment:
Looks like a Java long[2] is created for no reason here.  Looks like it existed before this PR, but would be good to get it fixed.

Reply:
```cudf::jni::native_jlongArray retTmp(env, 1);
    retTmp[0] = reinterpret_cast<jlong>(valid);
```
This seems like dead code to me? I am going to remove retTmp altogether and see what blows up

Reply:
done.